pipeline-runner/R/gem2s-2-load_user_files.R

```diff
@@ -239,6 +209,38 @@ parse_rhapsody_matrix <- function(config, input_dir) {
   return(list(counts_list = counts_list, annot = annot))
 }
 
+read_10x_annotations <- function(annot_fpath, sample) {
+  gene_column <- 1
+
+  annot <- read.delim(annot_fpath, header = FALSE)
+
+  # duplicate first col if there is only one col with gene names/ids, or
+  # if there is a "Gene Expression" col
+  if (ncol(annot) == 1 || annot[1, 2] == "Gene Expression") {
```

Comment:
`annot[1, 2] == "Gene Expression"` I'm having a hard time with this in R. The indexing `annot[1, 2]` means return the first and second column? how do you return first and second row?. In any case, it will return two values right? but you can directly compare it to a string? Is it going to make an element-wise comparison and then if any of those is true, return tru?

Reply:
annot[1,2] will return row 1, column 2 of the data frame. In this case we want to check if the second column is filled of elements named "Gene Expression" because that would mean that we have a column of gene ids/gene symbols and a column indicating the type of measurement. 

If we have less than 2 columns, that means we only have the feature symbols/ids, so it's safe to duplicate the first column as well. 

If we have more than one column or the second column is different from gene expression, that means we have two columns with ids/symbols and a third column with type of measurement.

All these cases are based on experience and cases we have actually seen in real life datasets.